hypertension/dc-7101-algorithms/py-root/lib/algorithms/bp_sufficiency.py

```diff
@@ -1,12 +1,109 @@
 from datetime import datetime
 from dateutil.relativedelta import relativedelta
+import operator
 
-from .utils import (
-    tally_diastolic_counts,
-    tally_systolic_counts,
-    calculate_reading_from_buckets,
-    bp_readings_meet_date_specs
-)
+def bp_readings_meet_date_specs(date_of_claim, bp_readings):
+    """
+    Determine from a list of BP readings if there exists two readings with a 1 month and 6 month date window 
+
+    :param date_of_claim: string of date of claim
+    :type date_of_claim: string
+    :param bp_readings: list of blood pressure readings
+    :type bp_readings: list
+    :return: boolean indicating if readings within the two date windows are present
+    :rtype: boolean
+    """
+
+    reading_within_one_month = False
+    reading_within_six_months = False
+    date_of_claim_date = datetime.strptime(date_of_claim, "%Y-%m-%d").date()
+
+    for reading in bp_readings:
+        bp_reading_date = datetime.strptime(reading["date"], "%Y-%m-%d").date()
+        one_month_difference = (date_of_claim_date - bp_reading_date).days <= 30
+        six_month_difference = (date_of_claim_date - bp_reading_date).days <= 180
+
+        if (one_month_difference and not reading_within_one_month):
+            reading_within_one_month = True
+        elif (one_month_difference and reading_within_one_month):
+            reading_within_six_months = True
+        elif (six_month_difference and not reading_within_six_months):
+            reading_within_six_months = True
+
+    return reading_within_one_month and reading_within_six_months
+
+def calculate_predominant_ratings(bp_readings):
```

Comment:
This is the new algorithm function that replaced the other three that were in the utilities file